src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,158 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import duke.exceptions.InvalidTaskException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * Holds an array of <code>Task</code> that actively keeps track of stored tasks in runtime.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructs a <code>TaskList</code> with no tasks.
+     */
+    protected TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a <code>TaskList</code> with specified task list.
+     *
+     * @param tasks <code>ArrayList</code> of tasks.
+     */
+    protected TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
```

Comment:
This is a good example of when to keep the this keyword! From the previous comment message